tests/functional/modules/test_zos_blockinfile_func.py

```diff
@@ -1157,7 +1150,7 @@ def test_ds_tmp_hlq_option(ansible_zos_module):
             hlq = result.get("stdout")
         if len(hlq) > 8:
             hlq = hlq[:8]
-        ds_full_name = hlq + "." + test_name.upper() + "." + ds_type
+        ds_full_name = hlq + "." + get_dataset(hosts)
```

Comment:
I think get_dataset currently gets you a ds name with an HLQ, what happens if get_dataset returns a data set name with the maximum name size allowed?  If we are doing this is better to also have a function that generates ds_names without HLQ.

That way in your data set name generator you can have:

1- Function to generate a random valid HLQ.
2- Function to generate a random valid ds name without HLQ.
3- Function that essentially uses 1 and 2 to generate a complete random valid ds name